packages/library-base/src/commonMain/kotlin/io/realm/kotlin/MutableRealm.kt

```diff
@@ -126,4 +126,26 @@ public interface MutableRealm : TypedRealm {
      * @throws IllegalArgumentException if the object is invalid, frozen or not managed by Realm.
      */
     public fun delete(deleteable: Deleteable)
+
+    /**
+     * Deletes all objects from this Realm.
+     */
+    public fun deleteAll()
+
+    /**
+     * Deletes all objects of the specified class from the Realm.
+     *
+     * @param KClass the class whose objects should be removed.
+     * @throws IllegalStateException if the class does not exist within the schema.
```

Comment:
It feels a bit weird that we are throwing `IllegalStateException` here. Especially since the DynamicRealm is thrown `IllegalArgumentException` I don't know if this is something we can change at the lower layers, but otherwise, we can catch the IllegalState, check the message and remap to `IllegalArgumentException` as needed. We are doing that elsewhere in a few places.